Resources/public/js/views/ez-dashboardblockbasedview.js

```diff
@@ -0,0 +1,217 @@
+         *
+         * @method _sortBlocks
+         * @protected
+         * @param blocks {Object} hash of blocks
+         * @return {Array} sorted list of blocks
+         */
+        _sortBlocks: function (blocks) {
+            var list = [];
+
+            Object.keys(blocks).forEach(function (key) {
+                list.push(blocks[key]);
+            });
+
+            list.sort(function (a, b) {
+                var priorityA = a.get('priority'),
+                    priorityB = b.get('priority');
+
+                if (priorityA === priorityB) {
+                    // sort alphabetically (from A to B)
+                    return a.get('identifier') < b.get('identifier');
+                } else {
+                    // sort by priority (max to 0)
+                    return priorityA > priorityB;
+                }
+            });
+
+            return list;
+        },
+
+        /**
+         * Verifies provided `blocks` attribute value
+         * by checking whether provided values are dashboard block view instances
+         *
+         * @method _checkBlocks
+         * @protected
+         * @param value {Any} new vaue of `blocks` attribute
+         * @return {Object}
+         */
+        _checkBlocks: function (value) {
+            var blocks = {};
+
+            if (typeof value !== 'object' || Array.isArray(value)) {
+                return blocks;
+            }
+
+            Object.keys(value).forEach(function (key) {
+                if (value[key] instanceof Y.eZ.BaseDashboardBlockView) {
+                    blocks[key] = value[key];
+                }
+            });
+
+            return blocks;
+        },
+
+        /**
+         * Adds a block to the dashboard
+         *
+         * @method addBlock
+         * @param view {Y.eZ.BaseDashboardBlockView} dashboard block view instance
+         * @return {Object} the hash of blocks
+         */
+        addBlock: function (view) {
+            var blocks = this.get('blocks'),
+                identifier = view.get('identifier');
+
+            if (!identifier) {
+                return blocks;
+            }
+
+            if (blocks[identifier]) {
+                blocks = this.removeBlock(identifier, true);
+            }
+
+            blocks[identifier] = view;
+
+            this.set('blocks', blocks);
+
+            return blocks;
+        },
+
+        /**
+         * Removes a block from the dashboard
+         *
+         * @method removeBlock
+         * @param identifier {String} block identifier
+         * @param preventUpdating {Boolean} should the `blocks` attribute value be updated?
+         * @return {Object} the hash of blocks
+         */
+        removeBlock: function (identifier, preventUpdating) {
+            var blocks = this.get('blocks'),
+                block = blocks[identifier];
+
+            block.removeTarget(this);
+            block.destroy({remove: true});
```

Comment:
IMHO, this should not be done in this method. for consistency this method should do exactly the opposite of `addBlock` so remove the block from the list and call `removeTarget`. The proper destruction of the block view is up to the user of this method.